### PR TITLE
MGMT-9279: Remove assisted-service database timeout

### PR DESCRIPTION
**Pull request by omertuc**
See commit message

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [x] On-Prem
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @vrutkovs

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

### CODE REVIEW

**Comment by vrutkovs** [comment]
+1 from me

File: cmd/main.go [hunk 1]
```diff
@@ -581,7 +581,7 @@ func setupDB(log logrus.FieldLogger) *gorm.DB {
 	var db *gorm.DB
 	var err error
 	// Tries to open a db connection every 2 seconds for up to 10 seconds.
-	timeout := 10 * time.Second
+	timeout := 30 * time.Second
```
Comment (mhrivnak) [inline cmd/main.go:584]: Why do we have a timeout at all? Why not just keep trying? A readiness check should be sufficient for communicating that the process is not ready.

**Review by mhrivnak — CHANGES_REQUESTED** [review]

File: cmd/main.go (hunk 1 above)
Comment (omertuc) [reply cmd/main.go:584]: @rwsu what do you think? You introduced it initially

Comment (omertuc) [reply cmd/main.go:584]: Created https://issues.redhat.com/browse/MGMT-9279 so we can discuss it further

Comment (mhrivnak) [reply cmd/main.go:584]: It's a small discussion to have, so I think we should just expedite it and remove the timeout unless someone can identify a reason not to.

Comment (omertuc) [reply cmd/main.go:584]: Fair enough, no harm in having a ticket to attach to this PR anyway. Let's wait for input from @rwsu

Comment (rwsu) [reply cmd/main.go:584]: The timeout can be removed. Before, the service would hard fail if the database wasn't ready. So the timeout was added to give more it more time to retry before failing.

Comment (omertuc) [reply cmd/main.go]: Ack, done

File: cmd/main.go [hunk 2]
```diff
@@ -578,9 +578,8 @@ func setupDB(log logrus.FieldLogger) *gorm.DB {
 	var db *gorm.DB
 	var err error
 	// Tries to open a db connection every 2 seconds for up to 10 seconds.
```
Comment (mhrivnak) [inline cmd/main.go:580]: Don't forget to fix this comment.

**Review by mhrivnak — CHANGES_REQUESTED** [review]
Other than the comment, LGTM.

**Review by mhrivnak — APPROVED** [review]

**Pull request merged**